runtime/js-compute-runtime/builtins/cache-simple.cpp

```diff
@@ -0,0 +1,440 @@
+  double ttl;
+  if (!JS::ToNumber(cx, ttl_val, &ttl)) {
+    return false;
+  }
+  if (ttl < 0 || std::isnan(ttl) || std::isinf(ttl)) {
+    JS_ReportErrorASCII(
+        cx, "SimpleCache.set: TTL parameter is an invalid value, only positive numbers can "
+            "be used for TTL values.");
+    return false;
+  }
+  options.max_age_ns = JS::ToUint64(ttl) *
+                       1'000'000'000; // turn second representation into nanosecond representation
+
+  JS::HandleValue body_val = args.get(1);
+  HttpBody source_body;
+  JS::UniqueChars buf;
+  size_t length;
+  JS::RootedObject body_obj(cx, body_val.isObject() ? &body_val.toObject() : nullptr);
+  // If the body parameter is a Host-backed ReadableStream we optimise our implementation
+  // by using the ReadableStream's handle directly.
+  if (body_obj && JS::IsReadableStream(body_obj)) {
+    if (RequestOrResponse::body_unusable(cx, body_obj)) {
+      JS_ReportErrorNumberASCII(cx, GetErrorMessage, nullptr,
+                                JSMSG_READABLE_STREAM_LOCKED_OR_DISTRUBED);
+      return false;
+    }
+
+    // If the stream is backed by a C@E body handle, we can use that handle directly.
+    if (builtins::NativeStreamSource::stream_is_body(cx, body_obj)) {
+      JS::RootedObject stream_source(cx,
+                                     builtins::NativeStreamSource::get_stream_source(cx, body_obj));
+      JS::RootedObject source_owner(cx, builtins::NativeStreamSource::owner(stream_source));
+      auto body = RequestOrResponse::body_handle(source_owner);
+      source_body = HttpBody(body.handle);
+    } else {
+      JS_ReportErrorNumberASCII(cx, GetErrorMessage, nullptr,
+                                JSMSG_SIMPLE_CACHE_SET_CONTENT_STREAM);
+      return false;
+    }
+
+    // The cache APIs require the length to be known upfront, we don't know the length of a
+    // stream upfront, which means the caller will need to supply the information explicitly for us.
+    if (!args.hasDefined(3)) {
+      JS_ReportErrorASCII(cx, "SimpleCache.set: length parameter is required when the value "
+                              "parameter is a ReadableStream. The length of the stream needs to be "
+                              "known before inserting into the cache.");
+      return false;
+    }
+
+    JS::HandleValue length_val = args.get(3);
+    double number;
+    if (!JS::ToNumber(cx, length_val, &number)) {
+      return false;
+    }
+    if (number < 0 || std::isnan(number) || std::isinf(number)) {
+      JS_ReportErrorASCII(
+          cx, "SimpleCache.set: length parameter is an invalid value, only positive numbers can "
+              "be used for length values.");
+      return false;
+    }
+    options.length = JS::ToInteger(number);
+  } else {
+    auto result = convertBodyInit(cx, body_val);
+    if (result.isErr()) {
+      return false;
+    }
+    std::tie(buf, length) = result.unwrap();
+    options.length = length;
+  }
+
```

Comment:
`length` doesn't get initialized in the `then` branch of this big conditional. How about moving its definition to after the conditional and setting it to `options.length`?
```suggestion
    std::tie(buf, options.length) = result.unwrap();
  }
  size_t length = options.length;
```